api/pkg/apis/v1alpha1/managers/activations/activations-manager.go

```diff
@@ -323,15 +331,35 @@ func (t *ActivationsManager) ReportStageStatus(ctx context.Context, name string,
 	return nil
 }
 
-func mergeStageStatus(activationState *model.ActivationState, current model.StageStatus) error {
+func mergeStageStatus(ctx context.Context, activationState *model.ActivationState, current model.StageStatus) error {
 	if current.Outputs["__site"] == nil {
 		// The StageStatus is triggered locally
 		if activationState.Status.StageHistory == nil {
 			activationState.Status.StageHistory = make([]model.StageStatus, 0)
 		}
+
+		currentBytes, _ := json.Marshal(current)
+		if len(currentBytes) > StageOutputMaxSize {
+			log.InfofCtx(ctx, "Stage %s size exceeds the limit, truncate the response body.", current.Stage)
+			outputLen := len(current.Outputs)
+			// outputLen can't be 0 since currentOutputSize > StageOutputMaxSize
+			for key, val := range current.Outputs {
+				valBytes, _ := json.Marshal(val)
+				if len(valBytes) > StageOutputMaxSize/outputLen {
+					current.Outputs[key] = valBytes[:StageOutputMaxSize/outputLen]
```

Comment:
Usually, you need to add ellipsis after the truncated parts.

Reply:
Agree. Will add it in the new version.